pkg/traceql/engine_metrics.go

```diff
@@ -1096,43 +1146,67 @@ const (
 	sumAggregation SimpleAggregationOp = iota
 	minAggregation
 	maxAggregation
+	avgAggregation
 )
 
 type SimpleAggregator struct {
 	ss               SeriesSet
 	exemplarBuckets  *bucketSet
 	len              int
-	aggregationFunc  func(existingValue float64, newValue float64) float64
+	aggregationFunc  func(b *SimpleAggregator, promLabel string, pos int, newValue float64)
 	start, end, step uint64
 	initWithNaN      bool
+
+	// Only for average
+	initAvg        bool
+	ssCounter      map[string]map[int]float64 // Counter of processed elements to calculate median
+	ssCompensation map[string]map[int]float64 // Avg compensation
 }
 
 func NewSimpleCombiner(req *tempopb.QueryRangeRequest, op SimpleAggregationOp) *SimpleAggregator {
 	l := IntervalCount(req.Start, req.End, req.Step)
 	var initWithNaN bool
-	var f func(existingValue float64, newValue float64) float64
+	var f func(b *SimpleAggregator, promLabel string, pos int, newValue float64)
```

Comment:
Since we need to modify more than one value on each iteration (counter and compensation) this does the trick